test/src/test/java/org/corfudb/integration/MergeTestListener.java

```diff
@@ -0,0 +1,222 @@
+package org.corfudb.integration;
+
+import com.google.protobuf.Message;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.LogReplicationListener;
+import org.corfudb.runtime.collections.*;
+import org.corfudb.test.SampleSchema;
+import org.junit.Before;
+
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.util.*;
+
+import static org.corfudb.integration.AbstractIT.*;
+import static org.corfudb.integration.LogReplicationAbstractIT.NAMESPACE;
+import static org.junit.Assert.fail;
+
+/**
+ * This class is for LogicalGroup E2E IT
+ *
+ *  (1) Open a merged table that store all the existing data and keys,
+ *      add keys to a set that store all the existing keys
+ *  (2) Start snapshot sync
+ *  (3) Process updates for snapshot sync, perform the diff by using
+ *      the set in (1) when processing replicated tables
+ *  (4) Snapshot sync ends, add the keys to the existing keys' set
+ *      and apply the keys to merged table
+ *  (5) Process updates for log entry sync
+ */
+@Slf4j
+public class MergeTestListener extends LogReplicationListener{
+
+    private final String corfuSingleNodeHost;
+    private final int corfuStringNodePort;
+
+    public CorfuRuntime runtime;
+
+    private final String singleNodeEndpoint;
+
+    private CorfuStore store;
+
+    private final String namespace = "test_namespace";
+
+    private final String userTableName = "data_table";
+
+    // Updates received through streaming
+    @Getter
+    private final LinkedList<CorfuStreamEntry> updates = new LinkedList<>();
+
+    private Set<Message> deletedKeys = new HashSet<>();
+
+    private Map<Message, CorfuStoreEntry> addedKeys = new HashMap<>();
+
+    private Set<Message> existingKeysInMergedTable = new HashSet<>();
+
+    Table<SampleSchema.Uuid, SampleSchema.ValueFieldTagOne, SampleSchema.Uuid> mergedTable;
+
+    MergeTestListener(CorfuStore corfuStore, String namespace) {
+        super(corfuStore, namespace);
+
+        corfuSingleNodeHost = PROPERTIES.getProperty("corfuSingleNodeHost");
+        corfuStringNodePort = Integer.valueOf(PROPERTIES.getProperty("corfuSingleNodePort"));
+        singleNodeEndpoint = String.format("%s:%d", corfuSingleNodeHost, corfuStringNodePort);
```

Comment:
I think current interface (constructor) introduces some assumptions, for example this test listener will be used in ITs like this:
https://github.com/CorfuDB/CorfuDB/blob/e5e660cf86a80e0b2f2cad827dad51d94feb3e3d/test/src/test/java/org/corfudb/integration/LogReplicationListenerIT.java#L274-L275

We now hard coded 1. Table schema 2. Table name (merge table name)
Please think about in ITs is there any case where we need to pass customized table names or schemas? Or if we want to hard code them here, should we expose them as public static fields such that ITs could align with them?